validator_client/src/check_synced.rs

```diff
@@ -1,82 +1,29 @@
-///                                         it is more than `SYNC_TOLERANCE` behind the highest
-///                                         known slot.
-///
-///  The second condition means the even if the beacon node thinks that it's syncing, we'll still
-///  try to use it if it's close enough to the head.
-pub async fn check_synced<T: SlotClock>(
+pub async fn check_node_health(
     beacon_node: &BeaconNodeHttpClient,
-    slot_clock: &T,
-    log_opt: Option<&Logger>,
-) -> Result<(), CandidateError> {
+    log: &Logger,
+) -> Result<(Slot, bool, bool), CandidateError> {
     let resp = match beacon_node.get_node_syncing().await {
         Ok(resp) => resp,
         Err(e) => {
-            if let Some(log) = log_opt {
-                warn!(
-                    log,
-                    "Unable connect to beacon node";
-                    "error" => %e
-                )
-            }
-
-            return Err(CandidateError::Offline);
-        }
-    };
-
-    // Default EL status to "online" for backwards-compatibility with BNs that don't include it.
-    let el_offline = resp.data.el_offline.unwrap_or(false);
-    let bn_is_synced = !resp.data.is_syncing || (resp.data.sync_distance.as_u64() < SYNC_TOLERANCE);
-    let is_synced = bn_is_synced && !el_offline;
-
-    if let Some(log) = log_opt {
-        if !is_synced {
-            debug!(
-                log,
-                "Beacon node sync status";
-                "status" => format!("{:?}", resp),
-            );
-
             warn!(
                 log,
-                "Beacon node is not synced";
-                "sync_distance" => resp.data.sync_distance.as_u64(),
-                "head_slot" => resp.data.head_slot.as_u64(),
-                "endpoint" => %beacon_node,
-                "el_offline" => el_offline,
+                "Unable connect to beacon node";
+                "error" => %e
             );
-        }
 
-        if let Some(local_slot) = slot_clock.now() {
-            let remote_slot = resp.data.head_slot + resp.data.sync_distance;
-            if remote_slot + 1 < local_slot || local_slot + 1 < remote_slot {
-                error!(
-                    log,
-                    "Time discrepancy with beacon node";
-                    "msg" => "check the system time on this host and the beacon node",
-                    "beacon_node_slot" => remote_slot,
-                    "local_slot" => local_slot,
-                    "endpoint" => %beacon_node,
-                );
-            }
+            return Err(CandidateError::Offline);
         }
-    }
+    };
 
-    if is_synced {
-        Ok(())
-    } else {
-        Err(CandidateError::NotSynced)
-    }
+    Ok((
+        resp.data.head_slot,
+        // Note that optimistic and EL status will both default to their healthy variants which may
+        // be undesirable.
+        resp.data.is_optimistic.unwrap_or(false),
+        resp.data.el_offline.unwrap_or(false),
```

Comment:
I think these two fields were `Option` for backward compatilbity reasons a while ago. Perhaps we can remove the `Option` from these two fields now they're required as part of the beacon API spec. 

https://github.com/ethereum/beacon-APIs/blob/860584e71109cc9dc34ab6b67c646ae9a818c07e/apis/node/syncing.yaml#L20

Reply:
Ooh good point. I've opened a separate PR for this => https://github.com/sigp/lighthouse/pull/6168